facade/src/main/java/org/jboss/pnc/facade/providers/BuildProviderImpl.java

```diff
@@ -599,6 +605,39 @@ public void setDependentArtifacts(String id, List<String> artifactIds) {
         repository.save(buildRecord);
     }
 
+    public Page<Build> getByAttribute(BuildPageInfo buildPageInfo, Map<String, String> attributeConstraints) {
+        Set<Predicate<BuildRecord>> predicates = new HashSet<>();
+        for (Map.Entry<String, String> entry : attributeConstraints.entrySet()) {
+            String key = entry.getKey();
+            String value = entry.getValue();
+            if (key.startsWith("!")) {
+                withoutAttribute(key);
+            } else {
+                withAttribute(key, value);
+            }
+        }
+
+        Predicate<BuildRecord>[] predicatesArray = predicates.toArray(new Predicate[predicates.size()]);
+
+        PageInfo pageInfo = toPageInfo(buildPageInfo);
+        SortInfo sortInfo = rsqlPredicateProducer.getSortInfo(type, buildPageInfo.getSort());
+        List<BuildRecord> resultList = ((BuildRecordRepository) BuildProviderImpl.this.repository)
+                .queryWithPredicatesUsingCursor(pageInfo, sortInfo, predicatesArray);
+
+        int hits = repository.count(predicatesArray);
+
+        return new Page<>(
+                buildPageInfo.getPageIndex(),
+                buildPageInfo.getPageSize(),
+                (int) Math.ceil((double) hits / buildPageInfo.getPageSize()),
```

Comment:
There should be new construcotr that does the page count computation, so you don't need to include this line.